modules/exploits/windows/local/ms16_075_reflection_juicy.rb

```diff
@@ -78,7 +77,8 @@ def initialize(info={})
         OptAddress.new('ListeningAddress', [ true, 'Set listening address for MITM DCOM communication', '127.0.0.1' ]),
         OptPort.new('ListeningPort', [ true, 'Set listening port for MITM DCOM communication', 7777 ]),
         OptString.new('LogFile', [ false, 'Set the log file' ])
-      ])
+      ]
+    )
   end
 
   def assign_target
```

Comment:
Looking beyond just the check() function, I think this entire assign_target() function is now pretty much dead code? It is called on line 172 where it is assigned to an unused variable (rubocop warns about it). It looks like this module has been previously converted to automatic detection. A quick test through removing this function and `rexploit`-ing seems to show it isn't needed either, but can someone please eyeball it and confirm there's no background magic involved in metasploit that might require it?

Reply:
I agree.  Looks like dead code after it got updated to automatic targeting and SYSTEM spawn in https://github.com/rapid7/metasploit-framework/pull/11230.  I'm for removing it here; you could put it in a commit alone to make reversion easier if it breaks everything down the road.

Reply:
Thanks for confirming! Code removed in f6e4b52.